segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/BaseMetadataStore.java

```diff
@@ -527,6 +538,138 @@ protected void setVersion(long version) {
         this.version.set(version);
     }
 
+    /**
+     * Helper class to encapsulates lock over group of keys.
+     */
+    private static class LockWrapper {
+        /**
+         * Global map to track active locks.
+         */
+        private static final HashMap<String, LockData> LOCKS = new HashMap<>();
```

Comment:
Why this is static?
With a 'static' holder we can have bad side effects, especially on tests.
Please consider using am instance field for this and so having an instance field for LockWrapper inside the enclosing class

Reply:
Good idea. I'll move it into outer class as instance member.